Mathlib/Data/Finset/Sum.lean

```diff
@@ -94,4 +96,83 @@ theorem disj_sum_strictMono_right (s : Finset α) :
     StrictMono (s.disjSum : Finset β → Finset (α ⊕ β)) := fun _ _ =>
   disjSum_ssubset_disjSum_of_subset_of_ssubset Subset.rfl
 
+lemma disjSum_inj {α β : Type*} {s₁ s₂ : Finset α} {t₁ t₂ : Finset β} :
+    s₁.disjSum t₁ = s₂.disjSum t₂ ↔ s₁ = s₂ ∧ t₁ = t₂ := by
+  simp [Finset.ext_iff]
+
+lemma Injective2_disjSum {α β : Type*} : Function.Injective2 (@disjSum α β) :=
+  fun _ _ _ _ => by simp [Finset.ext_iff]
+
+/--
+Given a finset of elements `α ⊕ β`, extract all the elements of the form `α`. This
+forms a quasi-inverse to `disjSum`, in that it recovers its left input.
+-/
+def toLeft (s : Finset (α ⊕ β)) : Finset α :=
+  s.disjiUnion (Sum.elim singleton (fun _ => ∅)) <| by
+    simp [Set.PairwiseDisjoint, Set.Pairwise, Function.onFun, eq_comm]
+
+/--
+Given a finset of elements `α ⊕ β`, extract all the elements of the form `β`. This
+forms a quasi-inverse to `disjSum`, in that it recovers its right input.
+-/
+def toRight (s : Finset (α ⊕ β)) : Finset β :=
+  s.disjiUnion (Sum.elim (fun _ => ∅) singleton) <| by
+    simp [Set.PairwiseDisjoint, Set.Pairwise, Function.onFun, eq_comm]
+
+@[simp] lemma mem_toLeft {s : Finset (α ⊕ β)} {x : α} : x ∈ s.toLeft ↔ inl x ∈ s := by
+  simp [toLeft]
+
+@[simp] lemma mem_toRight {s : Finset (α ⊕ β)} {x : β} : x ∈ s.toRight ↔ inr x ∈ s := by
+  simp [toRight]
+
```

Comment:
Do we have `Sum.swap`? If so, it would be good to show that `(s.map swap).toLeft = s.toRight` and vice versa.

Reply:
We have it as a map but not as an embedding, so I added the `.image` version of this.

Reply:
We have it as the Equiv, so I think you can get the `map` version from that.

Reply:
Added